charts/nri-metadata-injection/templates/admission-webhooks/job-patch/role.yaml

```diff
@@ -2,14 +2,14 @@
 apiVersion: rbac.authorization.k8s.io/v1
 kind: Role
 metadata:
-  name: {{ template "nri-metadata-injection.fullname" . }}-admission
+  name: {{ include "nri-metadata-injection.fullname.admission" . }}
   namespace: {{ .Release.Namespace }}
   annotations:
     "helm.sh/hook": pre-install,pre-upgrade,post-install,post-upgrade
     "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
   labels:
-    app: {{ template "nri-metadata-injection.name" $ }}-admission
-{{ include "nri-metadata-injection.labels" $ | indent 4 }}
+    {{- include "newrelic.common.labels" . | nindent 4 }}
```

Comment:
nit: i would switch the order and keep non optional things rendered first and after the optional , just for the health of my OCD

Reply:
I changes the order after [this comment](https://github.com/newrelic/k8s-metadata-injection/pull/163#discussion_r847224241) maybe I misunderstood... WDYT?

Reply:
interesting i'm more than curious ! I'm not compleatly understand this TBH

> Order matters because it will overwrite the map as it is finding labels.

@kang-makes could you explain why the order of this can break break things?

Reply:
As far as I understood there is a possibility that more than one pod with the same labels can exist at the same time: The job patching and the deployment. So we should need that selector labels from the deployment and job are different so they don't fight for the same resource.

This is the reason why, in this case, `app` is mandatory but only at `job` or `deployment` level. The rest of optional labels are... that... optional.

In case it is mandatory, I would use something like `app.kubernetes.io/component` (See https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels)

But I don't care about other kind of objects...

Reply:
Now I saw that I left that comment on a `Role` creating a really big confusion. Sorry 😅

Reply:
No problem, reordered here: [80d8784](https://github.com/newrelic/k8s-metadata-injection/pull/163/commits/80d8784d81d2faaf6bc5a6864be89799cde69df8)